src/orchard/xref.clj

```diff
@@ -13,21 +15,40 @@
     (symbol? thing) (var-get (find-var thing))
     (fn? thing) thing))
 
-(defn fn-deps
-  "Returns a set with all the functions invoked by `val`.
-  `val` can be a function value, a var or a symbol."
-  {:added "0.5"}
-  [val]
-  (let [val (as-val val)]
-    (set (some->> val class .getDeclaredFields
+(defn- fn-name [^java.lang.Class f]
+  (-> f .getName repl/demunge symbol))
+
+(defn- fn-deps-class
+  [v]
+  (let [^java.lang.Class v (if (class? v)
+                             v
+                             (eval v))]
+    (set (some->> v .getDeclaredFields
                   (keep (fn [^java.lang.reflect.Field f]
                           (or (and (identical? clojure.lang.Var (.getType f))
                                    (java.lang.reflect.Modifier/isPublic (.getModifiers f))
                                    (java.lang.reflect.Modifier/isStatic (.getModifiers f))
                                    (-> f .getName (.startsWith "const__"))
-                                   (.get f val))
+                                   (.get f (fn-name v)))
                               nil)))))))
 
+(defn fn-deps
+  "Returns a set with all the functions invoked by `val`.
+  `val` can be a function value, a var or a symbol."
+  {:added "0.5"}
+  [s]
+  (when-let [^clojure.lang.AFn v (as-val s)]
+    (let [f-class-name (-> v .getClass .getName)
+          ;; breaks when called with
+          ;; (.getDeclaredField clojure.lang.DynamicClassLoader "classCache")
+          classCache (->> clojure.lang.DynamicClassLoader .getDeclaredFields second)
```

Comment:
Why not type-hint this? Seems better than skipping Eastwood.

Reply:
If I type-hint this I get an IllegalAccess Exception, I'm guessing it's the same problem as the direct access.
I'll have a look at the bytecode produced by both versions to hopefully find out why one works and the other doesn't, but it's a lower priority than the other things.